Dockerfile

```diff
@@ -385,7 +385,8 @@ function install_airflow_dependencies_from_branch_tip() {
     pip install --root-user-action ignore \
       ${ADDITIONAL_PIP_INSTALL_FLAGS} \
       "https://github.com/${AIRFLOW_REPO}/archive/${AIRFLOW_BRANCH}.tar.gz#egg=apache-airflow[${AIRFLOW_EXTRAS}]" \
-      --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}" || true
+      --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}" \
+      || pip install 'pydantic>=2' || true
```

Comment:
Looks great. Yeah. Lets' remove it then and see if `pip` can resolve them on their own. If there are problems with "eager upgrade" and pip not finding the right resolution automatically, the right way to handle it will be to add "pydantic>=2" to `EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS` (currently empty so it should be commented that likely we should remove it in the future when the resolution is improved) in both Dockerfile and Dockerfile.ci. This is the way how we "help" `pip` by additionally limiting the `eager upgrade`.

Reply:
It would resolve it to 1.18 because (a dependency of) Moto pins it to `pydantic~=1.18`